internal/provider/resource_vpc_firewall_rules_test.go

```diff
@@ -611,3 +615,227 @@ func testAccFirewallRulesDestroy(s *terraform.State) error {
+	configV0 := fmt.Sprintf(`
+provider "oxide" {
+  host  = "%s"
+  token = "fake"
+}
+
+resource "oxide_vpc_firewall_rules" "test" {
+  vpc_id = "3fa85f64-5717-4562-b3fc-2c963f66afa6"
+  rules = [
+    {
+      action      = "allow"
+      name        = "single-protocol"
+      description = "Single protocol"
+      direction   = "inbound"
+      priority    = 65535
+      status      = "enabled"
+      filters = {
+        ports     = ["22"]
+        protocols = ["TCP"]
+      }
+      targets = [
+        {
+          type  = "subnet"
+          value = "default"
+        }
+      ]
+    },
+    {
+      action      = "allow"
+      name        = "multiple-protocols"
+      description = "Multiple protocols"
+      direction   = "inbound"
+      priority    = 65535
+      status      = "enabled"
+      filters = {
+        ports     = ["22"]
+        protocols = ["TCP", "UDP"]
+      }
+      targets = [
+        {
+          type  = "subnet"
+          value = "default"
+        }
+      ]
+    },
+    {
+      action      = "allow"
+      name        = "no-protocol"
+      description = "No protocol"
+      direction   = "inbound"
+      priority    = 65535
+      status      = "enabled"
+      filters = {
+        ports = ["22"]
+      }
+      targets = [
+        {
+          type  = "subnet"
+          value = "default"
+        }
+      ]
+    }
+  ]
+}
+`, ts.URL)
+
+	//lintignore:AT004 // Provider must connect to test server.
+	configV1 := fmt.Sprintf(`
+provider "oxide" {
+  host  = "%s"
+  token = "fake"
+}
+
+resource "oxide_vpc_firewall_rules" "test" {
+  vpc_id = "3fa85f64-5717-4562-b3fc-2c963f66afa6"
+  rules = [
+    {
+      action      = "allow"
+      name        = "single-protocol"
+      description = "Single protocol"
+      direction   = "inbound"
+      priority    = 65535
+      status      = "enabled"
+      filters = {
+        ports = ["22"]
+        protocols = [
+          { type : "tcp" }
+        ]
+      }
+      targets = [
+        {
+          type  = "subnet"
+          value = "default"
+        }
+      ]
+    },
+    {
+      action      = "allow"
+      name        = "multiple-protocols"
+      description = "Multiple protocols"
+      direction   = "inbound"
+      priority    = 65535
+      status      = "enabled"
+      filters = {
+        ports = ["22"]
+        protocols = [
+          { type : "tcp" },
+          { type : "udp" }
+        ]
+      }
+      targets = [
+        {
+          type  = "subnet"
+          value = "default"
+        }
+      ]
+    },
+    {
+      action      = "allow"
+      name        = "no-protocol"
+      description = "No protocol"
+      direction   = "inbound"
+      priority    = 65535
+      status      = "enabled"
+      filters = {
+        ports = ["22"]
+      }
+      targets = [
+        {
+          type  = "subnet"
+          value = "default"
+        }
+      ]
+    }
+  ]
+}
+`, ts.URL)
+
+	resource.UnitTest(t, resource.TestCase{
```

Comment:
I see this block is `resource.UnitTest` and the following block is `resource.Test`. The code inside is mostly the same. Should both blocks be the same?

Reply:
Ah good catch. I was testing something else and found out about `UnitTest`, which I think works best here. We also don't need the `PreventPostDestroyRefresh` anymore (I was trying to test the scenarios where we get an error, but Terraform doesn't like when the state and schema don't match).